packages/react-query-devtools/src/index.ts

```diff
@@ -1 +1,12 @@
-export * from './devtools'
+if (process.env.NODE_ENV !== 'development') {
```

Comment:
I find it's better to do `process.env.NODE_ENV === 'production'` since it matches with the behaviour of most tools (e.g. npm recognizes `production`) etc. and also what tools usually recommend to set for production (e.g. webpack). While most do set `development` in case it's not production, in case someone sets it to a different one (say for building with staging API etc.) then it'll cause different behaviour between other tools and React Query.

Reply:
we can do that, I just got this from the v3 implementation. Problem is that this doesn't really work with ESM. If we take a look at the codesandbox build:
- https://codesandbox.io/s/tanstack-query-example-react-basic-hofver

we can see the error:

```
Could not find module in path: '@tanstack/react-query-devtools/build/esm/devtools' relative to '/node_modules/@tanstack/react-query-devtools/build/esm/index.js'
```

I think the cjs builds are fine, and in the umd builds, I can see that it is properly there for dev builds and removed for prod builds. But I don't think this is how it can work with esm builds. Any ideas?

Reply:
@TkDodo afaik with webpack, we should be able to do something like this, in the package.json of devtools:

```js
  "exports": {
    "production": "./index-without-devtools.js",
    "default": "./index-with-devtools.js"
  }
```

https://webpack.js.org/guides/package-exports/

I haven't tried it, but looks like it would work.

Reply:
right, we used to have this in the v4 beta, but it got lost during the monorepo move. let me see if I can restore it.

Reply:
As per usual, esm 💩 -ing on the party. Let's try our best on the esm, but if we need to ship a version with the others fixed while we figure it out, we can.

Reply:
@sachinraja fyi

Reply:
@satya164 @TkDodo in case my comment gets lost, `process.env.NODE_ENV === 'production' will cause the dev tools to be run when running tests in test which uses 'test' for NODE_ENV